rules/catch-or-return.js

```diff
@@ -8,7 +8,7 @@
 
 const getDocsUrl = require('./lib/get-docs-url')
 const isPromise = require('./lib/is-promise')
-const isMemberCallWithObjectName = require('./lib/is-member-call-with-object-name')
+const isMemberCallWithObjectName = require('./lib/is-member-call-with-object-name.js')
```

Comment:
To be consistent it may be best to leave the extension off here :thinking: 

```suggestion
const isMemberCallWithObjectName = require('./lib/is-member-call-with-object-name')
```

---

This being said, I am in favour of enforcing extensions everywhere as it makes more sense to my brain loading the file, not an approximation of the file :)

Reply:
I've made the (conservative) change. I agree with your position, though, especially given the fact that browsers require the extension.

If someone wants to add extensions later (e.g., for browser support when ESLint may advertise itself as ready out of the box in the browser), I think we can do so in another PR.